File: a02_whirlpool/java/com/whirlpool/cooking/SplashFragment.java
package com.whirlpool.cooking;

import android.net.Uri;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.navigation.fragment.NavHostFragment;

import com.whirlpool.cooking.databinding.FragmentSplashScreenBinding;
import com.whirlpool.cooking.utils.CookingAppUtils;
import com.whirlpool.cooking.utils.EspressoIdlingResource;
import com.whirlpool.cooking.utils.SettingsUtils;
import com.whirlpool.cooking.utils.logging.HMILogHelper;
import com.whirlpool.hmi.cooking.viewmodel.CookingViewModelFactory;
import com.whirlpool.hmi.settings.SettingsViewModel;
import com.whirlpool.hmi.uicomponents.fragments.AbstractSplashFragment;
import com.whirlpool.hmi.uicomponents.tools.util.DisplayUtils;

import org.jetbrains.annotations.NotNull;


public class SplashFragment extends  AbstractSplashFragment {
    private FragmentSplashScreenBinding fragmentSplashScreenBinding;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        fragmentSplashScreenBinding = FragmentSplashScreenBinding.inflate(inflater);
        fragmentSplashScreenBinding.setLifecycleOwner(this);
        prepareVideoView();
        return fragmentSplashScreenBinding.getRoot();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        fragmentSplashScreenBinding = null;
    }

    private void prepareVideoView() {
        fragmentSplashScreenBinding.videoView.setVideoURI(Uri.parse("android.resource://" + requireContext().getPackageName() + "/" + R.raw.wp_logo_video));
    }

    @Override
    public void onViewCreated(@NonNull @NotNull View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        fragmentSplashScreenBinding.videoView.start();
        //ToDo:// Is there a better way to play loop, In kotlin there is a isloop parameter in video view
        fragmentSplashScreenBinding.videoView.setOnCompletionListener (mediaPlayer -> fragmentSplashScreenBinding.videoView.start());
    }

    @Override
    protected void setupApplication() {

    }

    /**
     * Method to release the media player during fragment callbacks
     */
    private void releasePlayer() {
        fragmentSplashScreenBinding.videoView.stopPlayback();
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        HMILogHelper.Logd("Started listening to SDK ready event - isReady()");
        // Register Idling Resource to wait for the simulator/ACU to respond with the model running
        if(BuildConfig.DEBUG) {
            HMILogHelper.Logd("Idling +++ ");
            EspressoIdlingResource.getInstance().increment();
        }
        CookingViewModelFactory.isReady().observe(this, isReady -> {
            // Start only when cooking is ready
            if (isReady) {
                HMILogHelper.Logd("Selected Product Variant is: " + CookingViewModelFactory.getProductVariantEnum());
                if(BuildConfig.DEBUG) {
                    HMILogHelper.Logd("Idling --- ");
                    EspressoIdlingResource.getInstance().decrement();
                }
                DisplayUtils.updateTheme(requireContext(), requireActivity(), CookingAppUtils.getCurrentTheme(SettingsViewModel.getSettingsViewModel()));
                releasePlayer();
            }
        });
    }


    @Override
    public void onDestroy() {
        CookingViewModelFactory.isReady().removeObservers(this);
        super.onDestroy();
    }
}

File: CookingApplianceM63KA/app/src/main/java/com/whirlpool/cooking/ka/ktn/android/presenter/fragments/SplashFragment.kt
package android.presenter.fragments

import android.net.Uri
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import com.whirlpool.cooking.common.utils.EspressoIdlingResource
import com.whirlpool.cooking.ka.BuildConfig
import com.whirlpool.cooking.ka.R
import com.whirlpool.cooking.ka.databinding.FragmentSplashBinding
import com.whirlpool.hmi.cooking.viewmodel.CookingViewModelFactory
import com.whirlpool.hmi.uicomponents.audio.WHRAudioManager
import core.utils.AppConstants
import core.utils.HMIExpansionUtils
import core.utils.HMILogHelper
import core.utils.SharedViewModel
import core.utils.UriUtils

/**
 * The class provides the launch screen which shows animated KitchenAid brand logo
 * until the SDK is initialized.
 */
class SplashFragment : Fragment() {
    private var splashBinding: FragmentSplashBinding? = null

    /**
     * Observe whether CookingViewModelFactory is initialized or not.
     */
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
    ): View {
        HMIExpansionUtils.disableFeatureHMIKeys(AppConstants.KEY_CONFIGURATION_DIGITAL_UNBOXING)
        splashBinding = FragmentSplashBinding.inflate(inflater, container, false)
        UriUtils.getUriFromRawResource(requireContext(), R.raw.video_brand_logo)
            ?.let { showVideoLogo(it) }
        observeCookingViewModel()
        sharedViewModel = ViewModelProvider(this)[SharedViewModel::class.java]
        return splashBinding!!.root
    }

    /**
     * Observe whether CookingViewModelFactory is initialized or not.
     */
    private fun observeCookingViewModel() {
        //Added for automation testing test cases.  Notify isReady() observe started callback
        if (BuildConfig.DEBUG) {
            HMILogHelper.Logd("Idling +++ ")
            EspressoIdlingResource.getInstance().increment()
        }
        CookingViewModelFactory.isReady().observe(
            viewLifecycleOwner
        ) { isReady: Boolean ->
            Log.e("isReady", "isReady = $isReady")
            checkIsCookingViewModelReady(isReady)
        }
    }

    private var sharedViewModel: SharedViewModel? = null

    /**
     * @param isReady - true/false status when observing the CookingViewModelFactory.
     */
    private fun checkIsCookingViewModelReady(isReady: Boolean) {
        try {
            if (isReady) {
                //Added for automation testing test cases. Notify isReady() observe success callback
                if (BuildConfig.DEBUG) {
                    HMILogHelper.Logd("Idling --- ")
                    EspressoIdlingResource.getInstance().decrement()
                }
                stopVideoPlayback()
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    /**
     * stops the video animation logo if it is playing already
     */
    private fun stopVideoPlayback() {
        try {
            if (splashBinding?.videoViewLogo?.isPlaying == true) {
                splashBinding?.videoViewLogo!!.stopPlayback()
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    /**
     * Used to load video file in the video view.
     * @param uri absolute location of the video file from the raw resource folder.
     */
    private fun showVideoLogo(uri: Uri) {
        try {
            splashBinding?.videoViewLogo?.setVideoURI(uri)
            splashBinding?.videoViewLogo?.setOnPreparedListener { mediaPlayer ->
                mediaPlayer.isLooping = true
            }
            splashBinding?.videoViewLogo?.setOnErrorListener { _, _, _ -> false }
            splashBinding?.videoViewLogo?.start()
            WHRAudioManager.getInstance().playAudio(context, R.raw.power_on)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    override fun onStop() {
        super.onStop()
        stopVideoPlayback()
    }

    override fun onDestroy() {
        splashBinding = null
        super.onDestroy()
    }
}